lib/consumer.js

```diff
@@ -176,19 +141,11 @@ Consumer.prototype.getThing = function(identity) {
  * @returns {Promise} that resolves with decoded thing schema as it is on blockchain at the moment.
  */
 Consumer.prototype.getSchema = function(index) {
-  var self = this;
-  return new Promise(function (fulfill, reject) {
-      self.registry.schemas.call(index, {from: self.address}, function(error, proto) {
-        if (error) {
-            console.error(error);
-            reject(error);
-            return;
-        }
-        if (proto.slice(0, 2) == '0x') {
-          proto = proto.slice(2);
-        }
-        fulfill(Schema.decodeHex(proto));
-    });
+  return this.registry.schemasAsync(index, {from: this.address}).then(function(proto) {
+    if (proto.slice(0, 2) == '0x') {
+      proto = proto.slice(2);
+    }
+    return Schema.decodeHex(proto);
```

Comment:
I know the es6 goodies, but the whole sdk is not following es6, so again it is out of scope of this task.